### ISSUE TITLE
Leading slash on image name causes confusion

### ISSUE THREAD

**Issue by larsks** [docker/docker#18558]
I have recently upgraded from docker 1.8.2 to docker 1.9.1.  I notice that docker seems to have changed the way it names images.  Previously, when building an image with:

    $ docker build -t larsks/someimage .

I would end up with:

    $ docker images | grep someimage
    larsks/someimage ...

But now I get:

    $ docker images | grep someimage
    /larsks/someimage ...

The problem with this is that `docker run larsks/someimage` will always pick the earlier image, without the leading `/`, which caused no end of confusion.

This behavior was unexpected and caused surprising behavior.

**Comment by thaJeztah** [comment]
Also related https://github.com/docker/docker/pull/18524 ?

**Comment by thaJeztah** [comment]
On second thought; Is this a vanilla docker version, or a version distributed by Red Hat? Can you add the information that @GordonTheTurtle mentioned above?

**Comment by pkgw** [comment]
I'll chime in and say that I see this issue, with resulting confusion, on a Red-Hat-provided docker. I’m pasting my info below.

A particular thing is that I can get in a state where `docker images` says that I have images named both `/foo` and `foo`. If I try to `docker rmi /foo`, it says that the image doesn't exist. If I `docker rmi foo`, first it deletes one image, then running it again deletes the second one. Confusing!

```
[root@iota ~]# docker version
Client:
 Version:         1.9.1-fc23
 API version:     1.21
 Package version: docker-1.9.1-4.git6ec29ef.fc23.x86_64
 Go version:      go1.5.1
 Git commit:      110aed2-dirty
 Built:           Wed Dec  9 09:09:16 UTC 2015
 OS/Arch:         linux/amd64

Server:
 Version:         1.9.1-fc23
 API version:     1.21
 Package version: docker-1.9.1-4.git6ec29ef.fc23.x86_64
 Go version:      go1.5.1
 Git commit:      110aed2-dirty
 Built:           Wed Dec  9 09:09:16 UTC 2015
 OS/Arch:         linux/amd64
[root@iota ~]# docker info
Containers: 1
Images: 24
Server Version: 1.9.1-fc23
Storage Driver: devicemapper
 Pool Name: docker-8:3-7753586-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: 
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 6.704 GB
 Data Space Total: 107.4 GB
 Data Space Available: 90.6 GB
 Metadata Space Used: 4.751 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.143 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.109 (2015-09-22)
Execution Driver: native-0.2
Logging Driver: journald
Kernel Version: 4.2.6-301.fc23.x86_64
Operating System: Fedora 23 (Workstation Edition)
CPUs: 8
Total Memory: 7.595 GiB
Name: iota.localdomain
ID: KFL4:R5YN:DBYB:WGFP:ELOW:B6IZ:UCBH:JG3Y:EPVD:GNRH:EYSP:SNKM
Username: pkgw
Registry: https://index.docker.io/v1/
[root@iota ~]# uname -a
Linux iota.localdomain 4.2.6-301.fc23.x86_64 #1 SMP Fri Nov 20 22:22:41 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by LK4D4** [comment]
Looks like not an issue in vanilla docker
